### DISCUSSION TITLE
I keep getting this: #<StandardError: Anthropic 400 Error

### DISCUSSION THREAD

**Discussion by alizjkrocks** [AllYourBot/hostedgpt#428] [General]
I keep getting this no matter what I type in and I'm not sure how to fix it...

#<StandardError: Anthropic 400 Error - invalid_request_error: There was an issue with the format or content of your request.>

**Comment by keithschacht** [comment]
@alizjkrocks That error is a new one. A bit more context would help me get to the bottom of it.

I assume you’ve entered your anthropic API key. Any other notable information about your setup?

Are you seeing this error appear in the actual chat response? It’s usually a much longer and potentially more informative error response. If you can post the whole thing, I might see other clues in that.

Have you tried OpenAI API key? I’m just curious if the issue is specific to anthropic (probably it is) or if there is something wrong with your overall setup.

**Reply by mitchfortunato** [reply]
@krschacht I'm getting the same issue as the above. I've tried creating a new key and resetting the key in HostedGPT and it's still not working. 
Base URL: https://api.anthropic.com/v1/
Token: my private token from the anthropic console. 
Billing: set up and funded. 

The chat stays stuck, where the small circle pulsates as though Sonnet is pondering. 

When finally I get a response, this is the chat response I'm receiving:

(I received a unexpected response from the API after retrying 3 times, "Anthropic 400 Error - invalid_request_error: There was an issue with the format or content of your request.". The AI servers may be experiencing trouble. Try again later or if you keep getting this error ensure your API key is valid and you haven''t run out of funds with your AI service.

It''s also helpful if you report this to the app developers at: https://github.com/allyourbot/hostedgpt/discussions)

:#<StandardError: Anthropic 400 Error - invalid_request_error: There was an issue with the format or content of your request.>

**Comment by lumpidu** [comment]
But one should also mention that Anthropic has in general higher elevated error rates. I am subscribed to their API-service mailing list and I get quite a few mails per week about the service being down, etc.

**Comment by zackgilbert** [comment]
I am trying to get a forked version up and running and receiving this same error:
<img width="767" alt="Screenshot 2025-04-08 at 2 56 02 PM" src="https://github.com/user-attachments/assets/e23e0b02-941a-446d-9e83-68cabe7821f8" />

**Reply by keithschacht** [reply]
Your discord ping prompted me to catch up on things and I see I missed this. I assume you got to the bottom of this by now. If you are able to get a fresh fork working, that would be a big help for me! Happy to take PRs and I'll work to give you quick response on things.

**Reply by zackgilbert** [reply]
Yeah, I should have a PR coming your way soon with some basic updates that'll include the latest Anthropic gem (0.4.0, I think?) getting up and running. I'll also have separate ones for updating the Ruby and Rails versions.
